scipy/stats/stats.py

```diff
@@ -4153,6 +4154,179 @@ def binary_search(n, n1, n2, side):
     return oddsratio, pvalue
 
 
+def _barnard_exact_compute_gammaln_combination(n):
+    """Compute all log combination of C(n, k)"""
+    gammaln_arr = special.gammaln(np.arange(n + 1) + 1)
+    return special.gammaln(n + 1) - gammaln_arr - gammaln_arr[::-1]
+
+
+def barnard_exact(
```

Comment:
Consider placing in `_hypotests.py`. `stats.py` is just getting too long.

Reply:
Moved in 1e4ceb2d182ae009c785bb7f59ed8270a53bdb61.